ibis/backends/dask/execution/generic.py

```diff
@@ -0,0 +1,390 @@
+def execute_binary_op_series_group_by(op, left, right, **kwargs):
+    left_groupings = left.grouper.groupings
+    right_groupings = right.grouper.groupings
+    if left_groupings != right_groupings:
+        raise ValueError(
+            'Cannot perform {} operation on two series with '
+            'different groupings'.format(type(op).__name__)
+        )
+    result = execute_binary_op(op, left.obj, right.obj, **kwargs)
+    return result.groupby(left_groupings)
+
+
+@execute_node.register(ops.BinaryOp, ddgb.SeriesGroupBy, simple_types)
+def execute_binary_op_series_gb_simple(op, left, right, **kwargs):
+    op_type = type(op)
+    try:
+        operation = constants.BINARY_OPERATIONS[op_type]
+    except KeyError:
+        raise NotImplementedError(
+            'Binary operation {} not implemented'.format(op_type.__name__)
+        )
+    else:
+        return left.apply(lambda x, op=operation, right=right: op(x, right))
+
+
+# TODO - grouping - #2553
+@execute_node.register(ops.BinaryOp, simple_types, ddgb.SeriesGroupBy)
+def execute_binary_op_simple_series_gb(op, left, right, **kwargs):
+    result = execute_binary_op(op, left, right, **kwargs)
+    return result.groupby(right.grouper.groupings)
+
+
+# TODO - grouping - #2553
+@execute_node.register(ops.UnaryOp, ddgb.SeriesGroupBy)
+def execute_unary_op_series_gb(op, operand, **kwargs):
+    result = execute_node(op, operand.obj, **kwargs)
+    return result
+
+
+# TODO - grouping - #2553
+@execute_node.register(
+    (ops.Log, ops.Round),
+    ddgb.SeriesGroupBy,
+    (numbers.Real, decimal.Decimal, type(None)),
+)
+def execute_log_series_gb_others(op, left, right, **kwargs):
+    result = execute_node(op, left.obj, right, **kwargs)
+    return result.groupby(left.grouper.groupings)
+
+
+# TODO - grouping - #2553
+@execute_node.register(
+    (ops.Log, ops.Round), ddgb.SeriesGroupBy, ddgb.SeriesGroupBy
+)
+def execute_log_series_gb_series_gb(op, left, right, **kwargs):
+    result = execute_node(op, left.obj, right.obj, **kwargs)
+    return result.groupby(left.grouper.groupings)
+
+
+@execute_node.register(ops.DistinctColumn, dd.Series)
+def execute_series_distinct(op, data, **kwargs):
+    return data.unique()
+
+
+@execute_node.register(ops.Union, dd.DataFrame, dd.DataFrame, bool)
+def execute_union_dataframe_dataframe(
+    op, left: dd.DataFrame, right: dd.DataFrame, distinct, **kwargs
+):
+    result = dd.concat([left, right], axis=0)
+    return result.drop_duplicates() if distinct else result
+
+
+@execute_node.register(ops.IfNull, simple_types, dd.Series)
+def execute_node_ifnull_scalar_series(op, value, replacement, **kwargs):
+    return (
+        replacement
+        if isnull(value)
+        else dd.from_pandas(
+            pd.Series(value, index=replacement.index), npartitions=1
+        )
+    )
+
+
+@execute_node.register(ops.NullIf, simple_types, dd.Series)
+def execute_node_nullif_scalar_series(op, value, series, **kwargs):
+    # TODO - not preserving the index
```

Comment:
I am curious why not preserving the index here? Does the input/output Series have the same index?

Reply:
Since we are doing a `da.where` on the `dd.Series.values` that will not carry the index information with it.